superset/views/core.py

```diff
@@ -2304,14 +2304,15 @@ def results(self, key):
         """Serves a key off of the results backend"""
         if not results_backend:
             return json_error_response("Results backend isn't configured")
-
         blob = results_backend.get(key)
         if not blob:
             return json_error_response(
                 'Data could not be retrieved. '
                 'You may want to re-run the query.',
                 status=410,
             )
+        if utils.is_prefetch_key(key):  # hack to not break when requesting prefetch
```

Comment:
Having the term "hack" in a comment is asking for trouble. Maybe we could rethink this.